test/e2e/multicast_test.go

```diff
@@ -0,0 +1,295 @@
+		for _, receiver := range mc.receiverConfigs {
+			for _, receiverMulticastInterface := range multicastInterfaces[receiver.idx] {
+				_, mRouteResult, _, err := RunCommandOnNode(nodeName(receiver.idx), fmt.Sprintf("ip mroute show to %s iif %s ", mc.group.String(), receiverMulticastInterface))
+				if err != nil {
+					return false, err
+				}
+				// If multicast traffic is sent from non-HostNetwork pods and senders-receivers are located in different nodes,
+				// the receivers should configure corresponding inbound multicast routes.
+				if mc.senderConfig.idx != receiver.idx && !receiver.isHostNetwork {
+					if len(mRouteResult) == 0 {
+						return false, nil
+					}
+				} else {
+					if len(mRouteResult) != 0 {
+						return false, nil
+					}
+				}
+				_, mAddrResult, _, err := RunCommandOnNode(nodeName(receiver.idx), fmt.Sprintf("ip maddr show %s | grep %s", receiverMulticastInterface, mc.group.String()))
+				if err != nil {
+					return false, err
+				}
+				// The receivers should also join multicast group.
+				// Note that in HostNetwork mode, the "join multicast" action is taken by mcjoin,
+				// which will not persist after mcjoin exits.
+				if !receiver.isHostNetwork {
+					if len(mAddrResult) == 0 {
+						return false, nil
+					}
+				} else {
+					if len(mAddrResult) != 0 {
+						return false, nil
+					}
+				}
+			}
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Error when waiting for multicast routes and stats: %v", err)
+	}
+	wg.Wait()
+}
+
+// computeMulticastInterfaces computes multicastInterfaces for each node.
+// It return map[int][]string with its key as node index and value multicastInterfaces.
+func computeMulticastInterfaces(t *testing.T, data *TestData) (map[int][]string, error) {
+	nodes, err := testData.clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	nodeIdxMcastIfaceMap := make(map[int][]string)
+	for _, node := range nodes.Items {
+		isControlPlaneNode := func() bool {
+			_, ok := node.Labels[labelNodeRoleControlPlane()]
+			return ok
+		}()
+		workerIdx := 1
+		var nodeIdx int
+		if isControlPlaneNode {
+			nodeIdx = 0
+		} else {
+			nodeIdx = workerIdx
+			workerIdx++
+		}
+		var externalInterfaces []string
+		externalInterfacesStr, err := data.GetNodeAnnotationByKey(node.Name, "external-interfaces")
+		if err != nil {
+			externalInterfaces = []string{}
+		} else {
+			err = yaml.Unmarshal([]byte(externalInterfacesStr), &externalInterfaces)
+			if err != nil {
+				externalInterfaces = []string{}
+			}
+		}
+		multicastInterfaces, err := data.GetMulticastInterfaces(antreaNamespace)
+		if err != nil {
+			return nil, err
+		}
+		// The final multicast interfaces used for the node is calculated by (externalInterfaceSet intersects multicastInterfaceSet adds transportInterface).
+		externalInterfaceSet := sets.NewString(externalInterfaces...)
+		multicastInterfaceSet := sets.NewString(multicastInterfaces...)
```

Comment:
Why we need to get the annotation and configmap at the same time?

Reply:
`multicastInterfaces` is a setting for multicast enabled Antrea cluster, as we discussed in https://github.com/antrea-io/antrea/pull/2835.

The annotation `external-interfaces` tells the people who write the tests that there are some other interfaces configured for the specified node in this testbed.

These two settings are designed to be unrelated. They happened to be used to compute the `actual interfaces used for forwarding multicast traffic for each node` here